File: FindMedian.py
# Find Median Number in List
# - Given a list of integers nums, return the median value.
# - Time: O(n log n), Space: O(1)

class Solution:
    def findMedian(self, nums: list[int]) -> float:
        if not nums:
            return None
        sorted_nums = sorted(nums)
        n = len(sorted_nums)
        mid = n // 2
        if n % 2 == 0:
            return (sorted_nums[mid - 1] + sorted_nums[mid]) / 2
        else:
            return sorted_nums[mid]

File: FindMedian.cs
// Find Median Number in List
// - Given a list of integers nums, return the median value.
// - Time: O(n log n), Space: O(1)

using System;
using System.Collections.Generic;
using System.Linq;

class Solution
{
    public double? FindMedian(List<int> nums)
    {
        if (nums == null || nums.Count == 0)
            return null; 

        nums.Sort(); 

        int n = nums.Count;
        int mid = n / 2;

        if (n % 2 == 0)
        {
            return (nums[mid - 1] + nums[mid]) / 2.0;
        }
        else
        {
            return nums[mid];
        }
    }
}
